### PR TITLE
Laboratory work #3, Pershina Olga - 21FPL1

### PR DESCRIPTION
**Pull request by OlyaP12**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
what about 6 or 8?

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -81,7 +91,7 @@ def preprocess_text(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens with no stop-words
         """
-        pass
+        return self._remove_stop_words(self._clean_and_tokenize(text))
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:94]: it is not clear. Save the result of `self._clean_and_tokenize(text)` to a variable and the use it to call `_remove_stop_words`

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -136,7 +149,13 @@ def encode(self, tokens: tuple[str, ...]) -> Optional[tuple[int, ...]]:
                 sequence of integer tokens
         In case of empty tokens input data, None is returned
         """
-        pass
+        if not tokens:
+            return None
+        self._learn_indices(tokens)
+        idx_of_words = []
+        for word in tokens:
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:156]: do this in one line

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -152,7 +171,11 @@ def decode(self, encoded_tokens: tuple[int, ...]) -> Optional[tuple[str, ...]]:
                 sequence of string tokens
         In case of out-of-dictionary input data, None is returned
         """
-        pass
+        for word in encoded_tokens:
+            if word not in self._id2word.keys():
+                return None
+        words = (self._id2word[token] for token in encoded_tokens)
+        return tuple(words)
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:178]: why do you need the `words` variable?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_keywords_textrank/main.py (hunk 1 above)
Comment (OlyaP12) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 2 above)
Comment (OlyaP12) [reply lab_3_keywords_textrank/main.py]: done

File: lab_3_keywords_textrank/main.py (hunk 3 above)
Comment (OlyaP12) [reply lab_3_keywords_textrank/main.py]: I realized that I could just write it in the return string without implementing a new variable, so I've done it

**Review by artyomtugaryov — APPROVED** [review]

**Pull request closed without merging**